service/src/main/java/com/fecklessweasel/service/datatier/CourseTable.java

```diff
@@ -46,4 +46,22 @@ public static int insertCourse(Connection conn, int univid, int deptid, int cour
             throw new ServiceException(ServiceStatus.DATABASE_ERROR, ex);
         }
     }
+    /**
+     *@author Hayden Schmackpfeffer
+     *@param conn SQL Connection
+     *@param deptid Department ID that all returned courses will have
+     *@return all tuples in Course table where deptid= deptid
+     */
+    public static ResultSet getAllWithDept(Connection conn, int deptid) throws ServiceException {
```

Comment:
I think this is correct how it is. It's searching the course table so it should be here.
We don't need an overload either. The department should provide it's own id when it calls the function.